addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/GuessParamsThread.java

```diff
@@ -0,0 +1,228 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import java.util.Date;
+import javax.swing.ListModel;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.model.ScanListenner;
+import org.zaproxy.zap.model.ScanThread;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.users.User;
+
+public class GuessParamsThread extends ScanThread implements GuesserListner {
+
+    private boolean stopScan = false;
+    private boolean isPaused = false;
+    private boolean isAlive = false;
+
+    private ExtensionParamMiner extension;
+    private int guessDone = 0;
+    private int guessTodo = 0;
+
+    private static final Logger log = LogManager.getLogger(GuessParamsThread.class);
+
+    private HttpMessage message;
+    private final String id;
+    private Date started;
+    private long timeTakenInMs;
+    private User scanUser;
+
+    private ParamGuesser paramGuesser;
+
+    public GuessParamsThread(
+            ExtensionParamMiner extension,
+            HttpMessage message,
+            String id,
+            String site,
+            ScanListenner scanListenner) {
+        super(site, scanListenner);
+        this.id = id;
+        this.extension = extension;
+        this.message = message;
+        this.site = site;
+        log.debug("GuessParamsThread created for " + site);
+    }
+
+    public long getTimeTakenInMs() {
+        if (this.timeTakenInMs > 0) {
+            return this.timeTakenInMs;
+        }
+        if (this.started != null) {
+            return System.currentTimeMillis() - started.getTime();
+        }
+        return 0;
+    }
+
+    @Override
+    public void run() {
+        try {
+            runScan();
+        } catch (Exception e) {
+            stopScan();
+        }
+    }
+
+    /** Runs the scan. */
+    private void runScan() {
+        // Do the scan
+        guessDone = 0;
+        started = new Date();
+        log.info("Starting parameter guess scan on {} at {}", site, started);
+        startGuesser();
+        this.isAlive = true;
+    }
+
+    private void startGuesser() {
+        // TODO Start a new ParamGuesser
+
+    }
+
+    @Override
+    public void stopScan() {
+        if (paramGuesser != null) {
+            paramGuesser.stop();
+        }
+        stopScan = true;
+        isAlive = false;
+        this.listenner.scanFinshed(site);
+    }
+
+    @Override
+    public void guesserProgress(int percentageComplete, int paramsTried, int totalParams) {
+        // TODO Auto-generated method stub
```

Comment:
All occurrences of this comment could be removed

Reply:
Yes, these will be removed as soon as I complete their Implementation. For some of the others, I'll add a TODO.